### PR TITLE
wayfire: 0.7.2 -> 0.7.5

### PR DESCRIPTION
**Pull request by wineee**
###### Description of changes

superseded https://github.com/NixOS/nixpkgs/pull/178093
superseded https://github.com/NixOS/nixpkgs/pull/203754

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/wayfire/default.nix [hunk 1]
```diff
@@ -1,33 +1,89 @@
...
+, wayland-protocols
+, wayland-scanner
+, wlroots
+, mesa
+, pango
+, xorg
+, libevdev
+, libxml2
+, glm
+, useSystemWlroots ? true
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/wayfire/default.nix:25]: Why do we need this option?

File: pkgs/applications/window-managers/wayfire/default.nix [hunk 2]
```diff
@@ -1,33 +1,89 @@
...
+    libinput
+    libjpeg
+    libxkbcommon
+    wayland-protocols
+    mesa
+    xorg.xcbutilwm
+    libevdev
+  ];
+
+  propagatedBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/wayfire/default.nix:64]: Do we really need to propagate all of them?

File: pkgs/applications/window-managers/wayfire/default.nix [hunk 3]
```diff
@@ -1,33 +1,89 @@
...
 
-  nativeBuildInputs = [ cmake meson ninja pkg-config wayland ];
-  buildInputs = [
-    cairo doctest libdrm libexecinfo libinput libjpeg libxkbcommon wayland
-    wayland-protocols wf-config wlroots mesa
+  nativeBuildInputs = [
+    cmake
+    meson
+    ninja
+    pkg-config
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/wayfire/default.nix:44]: Usually only 3 of them are required. I subject cmake could be removed, but not sure.

File: pkgs/applications/window-managers/wayfire/default.nix [hunk 4]
```diff
@@ -1,33 +1,89 @@
...
+    wlroots
+    wayland
+    libxml2
+  ];
+
+
+  mesonFlags = [
+    "--sysconfdir /etc"
+    (lib.mesonEnable "use_system_wlroots" useSystemWlroots)
+    "-Duse_system_wfconfig=disabled"
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/wayfire/default.nix:77]: We should use the system one if possible

File: pkgs/applications/window-managers/wayfire/wcm.nix [hunk 5]
```diff
@@ -1,20 +1,56 @@
...
+  version = "0.7.5";
 
-  src = fetchurl {
-    url = "https://github.com/WayfireWM/wcm/releases/download/v${version}/wcm-${version}.tar.xz";
-    sha256 = "19za1fnlf5hz4n4mxxwqcr5yxp6mga9ah539ifnjnqrgvj19cjlj";
+  src = fetchFromGitHub {
+    owner = "WayfireWM";
+    repo = pname;
+    rev = "v${version}";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/wayfire/wcm.nix:31]: Does https://github.com/luispabon/wdisplays not provide a buildable library?

File: pkgs/applications/window-managers/wayfire/wcm.nix [hunk 6]
```diff
@@ -1,20 +1,56 @@
...
+    sha256 = "sha256-LJR9JGl49o4O6LARofz3jOeAqseGcmzVhMnhk/aobUU=";
   };
 
-  nativeBuildInputs = [ meson ninja pkg-config wayland wrapGAppsHook ];
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    cmake
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/wayfire/wcm.nix:39]: Same here

File: pkgs/applications/window-managers/wayfire/wf-shell.nix [hunk 7]
```diff
@@ -1,23 +1,47 @@
...
-  # > Note to packagers: do not use the autogenerated "Source code"
-  # > archives from GitHub, but the wf-shell-0.4.0.tar.xz file.
-  src = fetchurl {
-    url = "https://github.com/WayfireWM/wf-shell/releases/download/v${version}/wf-shell-${version}.tar.xz";
-    sha256 = "1isybm9lcpxwyf6zh2vzkwrcnw3q7qxm21535g4f08f0l68cd5bl";
+  src = fetchFromGitHub {
+    owner = "WayfireWM";
+    repo = pname;
+    rev = "v${version}";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/wayfire/wf-shell.nix:24]: same here

File: pkgs/applications/window-managers/wayfire/wf-shell.nix [hunk 8]
```diff
@@ -1,23 +1,47 @@
...
+    sha256 = "sha256-iQUBuNjbZuf51A69RC6NsMHFZCFRv+d9XZ0HtP6OpOA=";
   };
 
-  nativeBuildInputs = [ meson ninja pkg-config wayland ];
+  nativeBuildInputs = [
+    meson
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/wayfire/wf-shell.nix:29]: same here

File: pkgs/applications/window-managers/wayfire/default.nix (hunk 3 above)
Comment (wineee) [reply pkgs/applications/window-managers/wayfire/default.nix]: meson use cmake to find `doctest`,   I tend to put them in `nativeCheckInputs`

File: pkgs/applications/window-managers/wayfire/wcm.nix (hunk 5 above)
Comment (wineee) [reply pkgs/applications/window-managers/wayfire/wcm.nix:27]: Wdisplays is not a library，it should be an optional runtime dependency.
Building wdisplays in wcm is unnecessary

File: pkgs/applications/window-managers/wayfire/default.nix (hunk 1 above)
Comment (wineee) [reply pkgs/applications/window-managers/wayfire/default.nix]: wlroots api is not stable，but for now, using system wlroots works fine

File: pkgs/applications/window-managers/wayfire/default.nix (hunk 2 above)
Comment (wineee) [reply pkgs/applications/window-managers/wayfire/default.nix:58]: It‘s `Requires` in wayfire.pc
glm is wf-config's propagatedBuildInputs (removed
at least wlroots should be propagated，it has different versions in nixpkgs，propagate wlroots could avoid using incompatible versions

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**